code/game/objects/structures/doors.dm

```diff
@@ -381,6 +402,12 @@
 	switching_states = FALSE
 	air_update_turf(TRUE)
 
+	if(autobump && last_bumper && src.Adjacent(last_bumper))
+		var/datum/lock/key/keylock = lock
+		if(istype(last_bumper.get_active_held_item(), /obj/item/key) || istype(last_bumper.get_active_held_item(), /obj/item/storage/keyring))
+			keylock.attack_wrap(src, last_bumper.get_active_held_item(), last_bumper, list(RIGHT_CLICK = "1"))
+		last_bumper = null
```

Comment:
This is ass, use /datum/element/connect_loc with COMSIG_ATOM_EXIT instead dm or @ me on the discord if you need help with that

Reply:
Or just no auto locking that's super minus soul of people forgetting to lock doors, auto unlocking I can see

Reply:
last_bumper is just bad if it has to be a thing it should be a weakref at least.

If it was to stay I would just make it so exiting the turf closes the door, make a callback with the time of the door closure passing in the user and force the user to attackby_secondary the door if adjacent. 

But it seems that the consensus is that its too much qol to auto unlock open and auto lock the door.